ui/components/AccountsNotificationPanel/AccountsNotificationPanelAccounts.tsx

```diff
@@ -632,15 +622,40 @@ export default function AccountsNotificationPanelAccounts({
 
   return (
     <div className="switcher_wrap">
+      <div className="account_actions_header">
+        {isEnabled(FeatureFlags.SUPPORT_CUSTOM_NETWORKS) && (
+          <AccountNetworkTabs selectedNetwork={selectedNetwork} />
+        )}
+        <AccountsSearchBar
+          searchAccountsValue={searchAccountsValue}
+          setSearchAccountsValue={setSearchAccountsValue}
+        />
+      </div>
+
       {accountTypes.map((accountType, switcherWrapIdx) => {
         const accountTypeTotals = accountTotals[accountType]
-
         // If there are no account totals for the given type, skip the section.
         if (accountTypeTotals === undefined || accountTypeTotals.length <= 0) {
           return <></>
         }
 
-        const accountTotalsByType = accountTypeTotals.reduce(
+        const filteredAccountTypeTotals = searchAccountsValue
+          ? accountTypeTotals?.filter(
+              (item: AccountTotal) =>
+                item.name
+                  ?.toLowerCase()
+                  .includes(searchAccountsValue.toLowerCase()) ||
+                item.address
+                  ?.toLowerCase()
+                  .includes(searchAccountsValue.toLowerCase()) ||
+                (item?.accountSigner?.type === "keyring" &&
+                  item.accountSigner?.shard
+                    ?.toLowerCase()
+                    .includes(searchAccountsValue.toLowerCase()))
+            )
+          : accountTypeTotals
+
+        const accountTotalsByType = filteredAccountTypeTotals.reduce(
```

Comment:
This applies to more than just this added logic, but there is a lot of logic in the return statement that can be handled outside of the return statement. This is okay for now but we will revisit this in the near future.